### ISSUE TITLE
Support for docker provider in vagrant 1.6

### ISSUE THREAD

**Issue by fzerorubigd** [vagrant-landrush/landrush#71]
Docker provider is part of vagrant now. its very nice if someone can add support for docker provider in landrush.

**Comment by electrofelix** [comment]
What I have noticed with enabling support is that landrush shows a concurrency issue in trying to start the dns server multiple times, which results in the following message on shutdown:
```
Stopping daemon...
Pid 7421 is not running. Has daemon crashed?
```
I suspect that the messages:
```
==> srv-5.ncs.net: [landrush] starting dns server
    srv-1.ncs.net: [landrush] 'ruby <homedir>/.vagrant.d/gems/gems/landrush-1.2.0/lib/landrush/start_server.rb 10053 <homedir>/.vagrant.d/data/landrush <homedir>/.vagrant.d/gems/gems'
==> srv-3.ncs.net: [landrush] starting dns server
==> srv-2.ncs.net: [landrush] starting dns server
    srv-1.ncs.net: [landrush] 'ruby <homedir>/.vagrant.d/gems/gems/landrush-1.2.0/lib/landrush/start_server.rb 10053 <homedir>/.vagrant.d/data/landrush <homedir>/.vagrant.d/gems/gems'
    srv-1.ncs.net: [landrush] 'ruby <homedir>/.vagrant.d/gems/gems/landrush-1.2.0/lib/landrush/start_server.rb 10053 <homedir>/.vagrant.d/data/landrush <homedir>/.vagrant.d/gems/gems'
==> srv-4.ncs.net: [landrush] starting dns server
    srv-1.ncs.net: [landrush] 'ruby <homedir>/.vagrant.d/gems/gems/landrush-1.2.0/lib/landrush/start_server.rb 10053 <homedir>/.vagrant.d/data/landrush <homedir>/.vagrant.d/gems/gems'
```
Indicate that the server config and launch code is iterated for each machine coming up in parallel because the check for whether the server is running is done on a per thread basis rather than per process.
However as this would be a common complication for any provider supporting parallel bring up, I'll look to log and address it as a separate issue, a simple per-process synchronization block and class variable for server status will resolve.

**Comment by hferentschik** [comment]
Merged via pull request #327
